app/core/uploader.py

```diff
@@ -37,26 +40,43 @@ async def upload_model(file: UploadFile = File(...)):
 
         await asyncio.sleep(1)
 
-        chip = ConfigManager().get().rknn_chip_type
-        asyncio.create_task(async_convert_model(file_path, chip))
-
-        return RedirectResponse(url="/upload_progress", status_code=303)
+        chip = ConfigManager().get().chip_type
+        if chip == ChipType.rk3588:
+            if suffix == ".pt":
+                asyncio.create_task(async_convert_model(file_path, chip))
+                return RedirectResponse(url="/upload_progress", status_code=303)
+            else:
+                return HTMLResponse("RKNN models should end with .pt for conversion!", status_code=400)
+        elif chip == ChipType.qcs6490:
+            if suffix == ".tflite":
+                return RedirectResponse("/", status_code=303)
+            else:
+                return HTMLResponse("TFLITE(QNN) models should be in TFLITE format!", status_code=400)
 
     except Exception as e:
         logger.exception("Model conversion failed", operation="upload")
         raise HTTPException(status_code=500, detail=f"Error converting model: {e}")
 
 
-def get_all_rknn_models() -> list[str]:
+def get_all_models() -> list[str]:
     try:
         if not UPLOAD_FOLDER.exists():
             return []
-
-        return [
-            name
-            for name in os.listdir(UPLOAD_FOLDER)
-            if name.endswith("rknn_model") and (UPLOAD_FOLDER / name).is_dir()
-        ]
+        config = ConfigManager().get()
+        if config.chip_type == ChipType.rk3588:
+            return [
+                name
+                for name in os.listdir(UPLOAD_FOLDER)
+                if name.endswith("rknn_model") and (UPLOAD_FOLDER / name).is_dir()
+            ]
+        elif config.chip_type == ChipType.qcs6490:
+            return [
+                name
+                for name in os.listdir(UPLOAD_FOLDER)
+                if name.endswith(".tflite") and (UPLOAD_FOLDER / name).is_file()
+            ]
```

Comment:
```suggestion
        if config.chip_type not in model_suffix:
            return []
        model_suffix = {ChipType.rk3588: ".pt", ChipType.qcs6490: ".tflite"}[config_chip_type]
        return [
            path
            for path in UPLOAD_FOLDER.iterdir()
            if path.is_file() and path.suffix
        ]
```
The model suffix getting should be done in the `ChipType` enum, it's used at least in one other place. You can add a small property getter there.
There, you could use a simple `if`, but using the dictionary makes it more expandable if we use other coprocessors in the future (not critical though).

Reply:
The thing is that the rknn model is a directory, while the tflite model is a file... The dictionary can have tuple values but i think at this point the if statements are more readable.

Reply:
ec4d63a77505a1ee6bcb8ad0fa22c77d4da00599